### ISSUE TITLE
Many SV variants from a VCF produced by SVDB are recognized as the same variant and not loaded

### ISSUE THREAD

**Issue by northwestwitch** [Clinical-Genomics/scout#3492]
For instance if you have a translocation from chrA:posA to chrB:posB and chrC:posC, only translocation from chrA:posA to chrB:posB is loaded as a variant

Of course this happens more often in cancer cases than RD cases

**Comment by northwestwitch** [comment]
Example provided by the cancer team:

```
14	106531322	SV_4005_1	N	]14:107010501]N	.	PASS	SVTYPE=BND;REGIONA=106531322,106531405;REGIONB=107010346,107010501;LFA=17,0;LFB=17,0;LTE=6,0;CTG=.	GT:cn:COV:DR:SR:LQ:RR:RD	0/1:2:30,27.162624934793953,41:6:0:0.0,0.0:0,26:5,33
14	106531322	SV_4006_1	N	]14:106712012]N	.	PASS	SVTYPE=BND;REGIONA=106531322,106531378;REGIONB=106712012,106712150;LFA=15,0;LFB=15,0;LTE=8,0;CTG=CCACATAATCTAAGTGGGACCTCAGCATTGAGCATTCATGGACATAAATGTGCGAATGATAGACACTGTGGACTGCTGGAGAGTGGAGGGAGGGGGTGATGGAATCTGGATTCCAAACCTCAGCATCACTCAATAATCCCATGTGACAAGTCCACACATATGCCCTCTGTATCTGAATGAAAACTTGAAATTAAATAAAAATCCTTATGTGAGAGCTGACTGGAAGCACCAAAGAGGACACTTGTTGTGGAGATTGACCTGCTCCTCATCCTAACTTAGGTGCTGGAGACAAATGTGTGCACATATGTC	GT:cn:COV:DR:SR:LQ:RR:RD	0/1:2:27,24.460027662517287,46:8:0:0.0,0.0:0,28:5,50
14	106712012	SV_4006_2	N	N[14:106531322[	.	PASS	SVTYPE=BND;REGIONA=106531322,106531378;REGIONB=106712012,106712150;LFA=15,0;LFB=15,0;LTE=8,0;CTG=CCACATAATCTAAGTGGGACCTCAGCATTGAGCATTCATGGACATAAATGTGCGAATGATAGACACTGTGGACTGCTGGAGAGTGGAGGGAGGGGGTGATGGAATCTGGATTCCAAACCTCAGCATCACTCAATAATCCCATGTGACAAGTCCACACATATGCCCTCTGTATCTGAATGAAAACTTGAAATTAAATAAAAATCCTTATGTGAGAGCTGACTGGAAGCACCAAAGAGGACACTTGTTGTGGAGATTGACCTGCTCCTCATCCTAACTTAGGTGCTGGAGACAAATGTGTGCACATATGTC	GT:cn:COV:DR:SR:LQ:RR:RD	0/1:2:27,24.460027662517287,46:8:0:0.0,0.0:0,28:5,50
14	107010501	SV_4005_2	N	N[14:106531322[	.	PASS	SVTYPE=BND;REGIONA=106531322,106531405;REGIONB=107010346,107010501;LFA=17,0;LFB=17,0;LTE=6,0;CTG=.	GT:cn:COV:DR:SR:LQ:RR:RD	0/1:2:30,27.162624934793953,41:6:0:0.0,0.0:0,26:5,33
```

**Comment by northwestwitch** [comment]
Examples: https://scout-stage.scilifelab.se/cust059/G1A1471p10_Balsamic and https://scout-stage.scilifelab.se/cust083/KMP-00064T-20191996305

**Comment by dnil** [comment]
Right, this situation will become untenable at some point. A suggestion would be to start adding a unique object index to the structural variants, and possibly add an extra step of detailed checking on fails against uniqueness to the current _id where we look also at the other fields.

**Comment by dnil** [comment]
So we remember it: for the overlap SNV-SV, @northwestwitch had the idea to add additional callers to the first variant, if the variants check out to be similar enough. This would be excellent there, but of course not solve the multiple-differerent-endpoint-bnds issue.

**Comment by dnil** [comment]
I currently do not understand though why we wouldn't get unique ids from the example above though, although I know it can happen with some callers. The (ALT, REF) field pairs all look unique?

**Comment by northwestwitch** [comment]
> I currently do not understand though why we wouldn't get unique ids from the example above though, although I know it can happen with some callers. The (ALT, REF) field pairs all look unique?

Unless we just use the chrom - start positions in the parsing

**Comment by dnil** [comment]
> > I currently do not understand though why we wouldn't get unique ids from the example above though, although I know it can happen with some callers. The (ALT, REF) field pairs all look unique?
> 
> Unless we just use the chrom - start positions in the parsing

It should be this, also for SVs?! https://github.com/Clinical-Genomics/scout/blob/6086937b4cb0da38fe52234a813fd14996d79a7e/scout/parse/variant/variant.py#L70

**Comment by dnil** [comment]
Without testing the variants at hand, at least in general parsing ALTs via cyvcf2 into db seems ok - see the `alternative` for this BND:
<img width="388" alt="Screenshot 2024-10-21 at 13 48 07" src="https://github.com/user-attachments/assets/2df1f4a5-1905-4b02-b9d5-4b30f354c09e">
